internal/e2etests/server/resource_test.go

```diff
@@ -916,6 +916,37 @@ func TestServerResource_Protection(t *testing.T) {
 	})
 }
 
+func TestServerResource_EmptySSHKey(t *testing.T) {
+	// Regression Test for https://github.com/hetznercloud/terraform-provider-hcloud/issues/727
+	var srv hcloud.Server
+
+	srvRes := &server.RData{
+		Name:    "server-empty-ssh-key",
+		Type:    e2etests.TestServerType,
+		Image:   e2etests.TestImage,
+		SSHKeys: []string{"\"\""},
```

Comment:
Why do we need this trick ?

Reply:
The template `r/hcloud_server.tf.tmpl` does not add any quotes to the values here. An empty string is rendered as exactly that, an empty string. We need to render our own quotes if we want them.

https://github.com/hetznercloud/terraform-provider-hcloud/blob/d57b38606c4b052b7d8181074d0860bd35935145/internal/testdata/r/hcloud_server.tf.tmpl#L16-L18